.profile.d/heroku-metrics-daemon.sh

```diff
@@ -5,63 +5,57 @@ if [[ -z "$HEROKU_METRICS_URL" ]] || [[ "${DYNO}" = run\.* ]]; then
     return 0
 fi
 
-export HEROKU_METRICS_PROM_ENDPOINT=${HEROKU_METRICS_PROM_ENDPOINT:-/metrics}
-export HEROKU_METRICS_PROM_PORT=$((PORT + 1))
+# heroku-metrics-agent.jar is added in bin/compile
+if [[ -f ${HOME}/bin/heroku-metrics-agent.jar ]]; then
+    if [[ -f build.sbt ]] || # Scala
+       [[ -d target/resolution-cache ]]; then # Scala (sbt-heroku)
+        export JAVA_OPTS="-javaagent:${HOME}/bin/heroku-metrics-agent.jar ${JAVA_OPTS}"
+    else
+        export JAVA_TOOL_OPTIONS="-javaagent:${HOME}/bin/heroku-metrics-agent.jar ${JAVA_TOOL_OPTIONS}"
+    fi
+else
+    export HEROKU_METRICS_PROM_ENDPOINT=${HEROKU_METRICS_PROM_ENDPOINT:-/metrics}
+    export HEROKU_METRICS_PROM_PORT=$((PORT + 1))
 
-STARTTIME=$(date +%s)
-BUILD_DIR=/tmp
+    STARTTIME=$(date +%s)
+    BUILD_DIR=/tmp
 
-DOWNLOAD_URL=$(curl --retry 3 -s https://agentmon-releases.s3.amazonaws.com/latest)
-if [ -z "${DOWNLOAD_URL}" ]; then
-    echo "!!!!! Failed to find latest agentmon. Please report this as a bug. Metrics collection will be disabled this run."
-    return 0
-fi
+    DOWNLOAD_URL=$(curl --retry 3 -s https://agentmon-releases.s3.amazonaws.com/latest)
+    if [ -z "${DOWNLOAD_URL}" ]; then
+        echo "!!!!! Failed to find latest agentmon. Please report this as a bug. Metrics collection will be disabled this run."
+        return 0
+    fi
 
-BASENAME=$(basename "${DOWNLOAD_URL}")
+    BASENAME=$(basename "${DOWNLOAD_URL}")
 
-curl -L --retry 3 -s -o "${BUILD_DIR}/${BASENAME}" "${DOWNLOAD_URL}"
+    curl -L --retry 3 -s -o "${BUILD_DIR}/${BASENAME}" "${DOWNLOAD_URL}"
 
-# Ensure the bin folder exists, if not already.
-mkdir -p "${BUILD_DIR}/bin"
+    # Ensure the bin folder exists, if not already.
+    mkdir -p "${BUILD_DIR}/bin"
 
-# Extract agentmon release
-tar --warning=no-unknown-keyword -C "${BUILD_DIR}/bin" -zxf "${BUILD_DIR}/${BASENAME}"
-chmod +x "${BUILD_DIR}/bin/agentmon"
+    # Extract agentmon release
+    tar --warning=no-unknown-keyword -C "${BUILD_DIR}/bin" -zxf "${BUILD_DIR}/${BASENAME}"
+    chmod +x "${BUILD_DIR}/bin/agentmon"
 
-ELAPSEDTIME=$(($(date +%s) - STARTTIME))
-echo "agentmon setup took ${ELAPSEDTIME} seconds"
+    ELAPSEDTIME=$(($(date +%s) - STARTTIME))
+    echo "agentmon setup took ${ELAPSEDTIME} seconds"
 
-AGENTMON_FLAGS=()
+    AGENTMON_FLAGS=()
```

Comment:
May as well just make this `AGENTMON_FLAGS=("-statsd-addr=:${PORT}")` and remove the line below.

Reply:
Fixed in #30